### DISCUSSION TITLE
Default findOne queries broken when migrating from beta.19.3 to beta.19.5, beta.20.x

### DISCUSSION THREAD

**Discussion by nurikabe** [strapi/strapi#6232] [Legacy Topics]
**Describe the bug**

MongoDB `findOne` default handlers always return the first record.

For example, in `api/document/config/routes.json`:

```
{
      "method": "GET",
      "path": "/documents/:_id",
      "handler": "Document.findOne",
      "config": {
        "policies": []
      }
    },
```

(Nothing overriding this in `api/document/controllers`.)

This handler always returns the first record.

**Expected behavior**

As in  beta.19.3, routes.json handlers should return a record for the given id.

**System**

- Node.js version: v12.13.1
- NPM version: 6.10.3
- Strapi version: v3.0.0-beta.20.3
- Database: MongoDB
- Operating system: Alpine Linux

**Comment by nurikabe** [comment]
Also note, we used to be able to do:

```
const document = await strapi.services.document.findOne(ctx.params);
```

However, this now throws a `Your filters contain a field '0' that doesn't appear on your model definition nor it's relations` error.

Has Strapi's ability to auto-deduce the query from `ctx.params` changed?

**Comment by lauriejim** [comment]
Hello! How can I reproduce to help you?
I'm not sure to understand :/

**Comment by derrickmehaffy** [comment]
Please see the following migration guide: https://strapi.io/documentation/3.0.0-beta.x/migration-guide/migration-guide-beta.19-to-beta.19.4.html#route-syntax-changes
